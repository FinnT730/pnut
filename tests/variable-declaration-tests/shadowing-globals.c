int x = 1, y, z = 2;

void putstring(char *s) {
  while (*s) {
    putchar(*s);
    s = s + 1;
  }
}

void testShadowingGlobals() {
  int x = 0, z = 3;
  y = 4;
  putchar(x + 48);
  putchar(10);
  putchar(y + 48);
  putchar(10);
  putchar(z + 48);
  putchar(10);
}

int main() {
  putstring("Test Shadowing Globals:\n");
  testShadowingGlobals();
  putstring("Global Variables:\n");
  putchar(x + 48);
  putchar(10);
  putchar(y + 48);
  putchar(10);
  putchar(z + 48);
  putchar(10);

  return 0;
}