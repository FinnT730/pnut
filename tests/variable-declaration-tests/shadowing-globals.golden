Test Shadowing Globals:
0
4
3
Global Variables:
1
4
2
